cylc/flow/platforms.py

```diff
@@ -421,8 +421,28 @@ def get_install_target_from_platform(platform):
     return platform.get('install target')
 
 
-def is_platform_with_target_in_list(
-        install_target, distinct_platforms_list):
+def get_install_target_to_platforms_map(platform_names):
+    """Get a dictionary of unique install targets and the platforms which use
+    them.
+
+    Return {install_target_1: [platform_1_dict, platform_2_dict, ...], ...}
+
+    Args:
+        platform_names (list): List of platform names to look up in the
+            global config.
+    """
+    platform_names = set(platform_names)
+    platforms = [get_platform(p_name) for p_name in platform_names]
+    install_targets = set(get_install_target_from_platform(platform)
```

Comment:
Do you need this second `set`?

Reply:
I think yes because otherwise you would end up with duplicate install targets.

Aside, Python has now given us set comprehensions to match list and dict comprehensions, but we don't get tuple comprehensions, those map to generators.

```
install_targets = {
    get_install_target_from_platform(platform)
    for platform in platforms
}
```